index.ts

```diff
@@ -0,0 +1,11 @@
+import {Readable} from "stream";
+
+export function streamifyArray(array: unknown[]) {
+   const readable = new Readable({objectMode: true});
+   readable._read = function (size: number) {
+      for (let i = 0; i < size; i++) {
```

Comment:
The old version didn't have this for-loop. I think we don't need it?

Reply:
`Array.prototype.shift()` doesn't take a parameter, so was even wondering how it has worked until now.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/shift

Reply:
Oh no...
This was probably supposed to be `splice()` then.

The old code probably didn't cause any problems though, as it just lead to every `_read` only emitting one value at a time.

Could you try with `splice` instead of the for-loop? *Should* be a bit more efficient (and easier code-wise).

Reply:
Definitely! However, then an array will be pushed to `readable`, is this as intended? With a for-loop, for every i in size, an element is pushed to `readable`, so the two behave differently. Not sure what the expected behaviour should be.

Reply:
Ah yes, you're right, you can't push an array into a readable.
The code as-is should be fine them :-)